lib/kb_Metrics/metricsDBs.py

```diff
@@ -34,9 +34,12 @@ def __init__(self, mongo_host, mongo_dbs, mongo_user, mongo_psswd):
         self.metricsDBs = dict()
         for m_db in mongo_dbs:
             # create the client and authenticate
-            self.mongo_clients[m_db] = MongoClient(
-                "mongodb://" + mongo_user + ":" + mongo_psswd +
-                "@" + mongo_host + "/" + m_db)
+            if mongo_host:
+                self.mongo_clients[m_db] = MongoClient(
+                    "mongodb://" + mongo_user + ":" + mongo_psswd +
+                    "@" + mongo_host + "/" + m_db)
+            else:  # default to localhost:27017
+                self.mongo_clients[m_db] = MongoClient()
```

Comment:
you don't need this if_else structure.

Reply:
I will remove this and test.

Reply:
Tested after I removed the if_else, the tests failed without if_else.

Reply:
So you should fix your tests.

Reply:
I have to test the funcs in the metricsDBs.py module, for that the __init__ needs mongo connection params to be passed create the metricsDBs object.

Reply:
Just pass mongo_host = ‘localhost’

Reply:
For in-container mongodb connection, host is not required.  But for remote connections it is required.  Therefore in the __init__ func for MetricsMongoDBI, host is checked with if_else.

Reply:
I don’t get what you mean. You shouldn’t add unnecessary logic just for local tests. If you can’t pass mongo_host as localhost. Just mock your func.

Reply:
That's assuming we always will pass non-empty host.  Mongo will automatically treat it as 'localhost' if '' is passed.  I don't think that is unnecessary logic.